connection.go

```diff
@@ -170,7 +176,16 @@ func (c *Connection) Connect() error {
 
 	c.run()
 
-	if c.Opts.OnConnect != nil {
+	if c.Opts.OnConnectCtx != nil {
```

Comment:
@bpross I think about two options here.

1. Get rid of `Options/OnConnect` and for backward compatibility, we can update the `OnConnect` function:
```go
func OnConnect(h func(c *Connection) error) Option {
	return func(opts *Options) error {
		opts.OnConnectCtx = func(_ context.Context, c *Connection) error {
                  return h(c)
                }
		return nil
	}
}
```

this is not 100% backward compatible, as if someone directly sets `Options/OnConnect` then the code will be broken. While no one should directly work with this option, it's exportable and we have no control over it now.

2. We can change the code to something like:
```go
	onConnect := c.Opts.OnConnectCtx
	if onConnect == nil && c.Opts.OnConnect != nil {
		onConnect = func(_ context.Context, c *Connection) error {
			return c.Opts.OnConnect(c)
		}
	}

	if onConnect != nil {
		if err := onConnect(ctx, c); err != nil {
			...
```

Reply:
I think the second of your options is preferable and cleans up the code a bit. It reduces branch complexity and keeps it backwards compatible if someone is directly setting `OnConnect`